client/client.go

```diff
@@ -402,7 +406,49 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	return client.doSyncWithOpts(method, path, query, headers, body, v, nil)
 }
 
+func (client *Client) checkMaintenanceJSON() error {
+	f, err := os.Open(dirs.SnapdMaintenanceFile)
+	// just continue if we can't read the maintenance file and
+	// it does exist (i.e. perm error)
+	if err == nil {
```

Comment:
Given that we don't check the actual error, it'd be easier to read if this did:
```
if err != nil {
   return nil
}
```
and the rest was unindented one level.